p1-staking-pool-dyn/src/lib.rs

```diff
@@ -218,13 +209,15 @@ impl Contract {
     /// Panics if user has not staked anything.
     #[payable]
     pub fn withdraw_crop(&mut self) -> Promise {
-        let (aid, mut vault) = self.get_vault();
+        let aid = env::predecessor_account_id();
+        let mut vault = self.get_vault_or_default(&aid);
```

Comment:
ditto, we should panic if account doesn't exist.

Reply:
it will fail because rewards==0